client/components/LearnerViewDirective.js

```diff
@@ -477,10 +477,22 @@ tie.directive('learnerView', [function() {
             feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
               additionalHeightForLoadingIndicator;
             $timeout(function() {
-              SolutionHandlerService.processSolutionAsync(
-                tasks[currentTaskIndex], code,
+              var index = 0;
+              var fn = function() {
+                SolutionHandlerService.processSolutionAsync(
+                tasks[index], code,
                 question.getAuxiliaryCode(language), language
-                ).then(setFeedback);
+                ).then(function(feedback) {
+                  if (index === currentTaskIndex ||
+                      !feedback.isAnswerCorrect()) {
+                    setFeedback(feedback);
```

Comment:
1. The following code fails on the first task:
![screenshot from 2017-04-10 15 11 57](https://cloud.githubusercontent.com/assets/6829866/24886423/57ba8c38-1e09-11e7-81aa-1d760a6b8ba2.png)

2. Now it passes
![screenshot from 2017-04-10 15 10 25](https://cloud.githubusercontent.com/assets/6829866/24886444/751add6e-1e09-11e7-820e-6fe70b72a81e.png)

3. Such code passes the second task, but fails on the first one. Originally the system still approves it, whereas with our code, the feedback from the first test is displayed:
![screenshot from 2017-04-10 15 11 23](https://cloud.githubusercontent.com/assets/6829866/24886488/baf64710-1e09-11e7-993b-c7d32b04818b.png)

4. If we write down a code that passes the first task but fails on the second one, the system will notify us again:
![screenshot from 2017-04-10 15 10 56](https://cloud.githubusercontent.com/assets/6829866/24886505/d4bb790e-1e09-11e7-8c49-cb4ab747f92a.png)